### PR TITLE
Add workaround for QIP IMAP server. This server returns non-expected …

### PR DESCRIPTION
**Pull request by disaykin**
…response to storeFlags command.

### CODE REVIEW

File: src/low-level/imap/mailimap.c [hunk 1]
```diff
@@ -2778,3 +2779,14 @@ LIBETPAN_EXPORT
 int mailimap_is_rambler_workaround_enabled(mailimap * session) {
   return session->is_rambler_workaround_enabled;
 }
+
+LIBETPAN_EXPORT
+void mailimap_set_qip_workaround_enabled(mailimap * session, int enabled) {
+  if (session)
```
Comment (dinhviethoa) [inline src/low-level/imap/mailimap.c]:
```
if (session) {
  session->is_qip_workaround_enabled = enabled;
}
```

**Comment by dinhviethoa** [comment]
Weird that the build is failing on macOS :(

**Comment by disaykin** [comment]
I see the problem with building of sasl library already fixed. Thanks!

File: src/low-level/imap/mailimap_parser.c [hunk 2]
```diff
@@ -7713,6 +7713,14 @@ mailimap_msg_att_parse_progress(mailstream * fd, MMAPString * buffer, struct mai
                                                  body_progr_fun, items_progr_fun,
                                                  context, msg_att_handler, msg_att_context);
   if (r != MAILIMAP_NO_ERROR) {
+    if (mailimap_parser_context_is_qip_workaround_enabled(parser_ctx)) {
+      r = mailimap_cparenth_parse(fd, buffer, parser_ctx, &cur_token);
+      if (r == MAILIMAP_NO_ERROR) {
+        // QIP returns "* num FETCH ()" response for storeFlags operations.
```
Comment (dinhviethoa) [inline src/low-level/imap/mailimap_parser.c]: Could you create an empty list instead of list == NULL?

File: src/low-level/imap/mailimap_types.c [hunk 3]
```diff
@@ -1732,9 +1732,11 @@ mailimap_msg_att_new(clist * att_list)
 LIBETPAN_EXPORT
 void mailimap_msg_att_free(struct mailimap_msg_att * msg_att)
 {
-  clist_foreach(msg_att->att_list,
-      (clist_func) mailimap_msg_att_item_free, NULL);
-  clist_free(msg_att->att_list);
+  if (msg_att->att_list != NULL) {
```
Comment (dinhviethoa) [inline src/low-level/imap/mailimap_types.c]: It would prevent to do this and break all potential other assumptions around.

Comment (disaykin) [inline src/low-level/imap/mailimap_types.c]: Done

File: src/low-level/imap/mailimap_parser.c (hunk 2 above)
Comment (disaykin) [inline src/low-level/imap/mailimap_parser.c]: Done

**Comment by dinhviethoa** [comment]
Thanks @disaykin!

**Pull request merged**